transcendence-app/src/game/game.gateway.ts

```diff
@@ -15,99 +16,47 @@ import { Socket, Server } from 'socket.io';
 import { BadRequestTransformationFilter } from '../shared/filters/bad-request-transformation.filter';
 import { TwoFactorAuthenticatedGuard } from '../shared/guards/two-factor-authenticated.guard';
 import { GameInputDto } from './dto/game-input.dto';
-import {
-  GameState,
-  paddleMoveRight,
-  paddleMoveLeft,
-  paddleStop,
-  runGameFrame,
-  newGame,
-  GameStatus,
-} from 'pong-engine';
-import { OnEvent } from '@nestjs/event-emitter';
-import { GamePairing } from './infrastructure/db/game-pairing.entity';
+import { GameService } from './game.service';
 
-type UserId = string;
-const FPS = 60;
-const DELTA_TIME = 1 / FPS;
 @WebSocketGateway({ path: '/api/v1/socket.io' })
 @UseGuards(TwoFactorAuthenticatedGuard)
 @UseInterceptors(ClassSerializerInterceptor)
 @UseFilters(BadRequestTransformationFilter)
 export class GameGateway {
   @WebSocketServer() server!: Server;
-  games = new Map<UserId, GameState>();
-  intervalId: NodeJS.Timer | undefined = undefined;
-
-  runServerGameFrame() {
-    this.games.forEach((gameState: GameState, userId: UserId) => {
-      const state = runGameFrame(DELTA_TIME, gameState);
-      this.games.set(userId, state);
-    });
-  }
 
-  sendGamesUpdates() {
-    this.games.forEach((gameState: GameState, userId: UserId) => {
-      this.sendGameUpdate(gameState, userId);
-    });
-  }
+  constructor(private readonly gameService: GameService) {}
 
-  sendGameUpdate(gameState: GameState, userId: UserId) {
-    this.server.to(userId).emit('updateGame', gameState, Date.now());
+  afterInit(server: Server) {
+    this.gameService.server = server;
   }
 
   @SubscribeMessage('joinGame')
-  handleGameJoin(@ConnectedSocket() client: Socket) {
-    const userId = client.request.user.id;
-    // just intended to simulate wait, to be deleted
-    setTimeout(() => {
-      if (!this.games.has(userId)) {
-        this.games.set(userId, newGame());
-        if (!this.intervalId) {
-          // buscar una mejor manera de hacer esto -> instanciar un gameRunner?
-          this.intervalId = setInterval(() => {
-            this.runServerGameFrame();
-            this.sendGamesUpdates();
-          }, DELTA_TIME * 1000);
-        }
-      }
-      this.server.to(userId).emit('joinGame', { res: 'ok' });
-    }, 2000);
+  handleGameJoin(
+    @ConnectedSocket() client: Socket,
+    @MessageBody('gameRoomId', ParseUUIDPipe) gameRoomId: string,
+  ) {
+    this.gameService.handleGameJoin(client, gameRoomId);
   }
 
   @SubscribeMessage('leaveGame')
```

Comment:
I have to say that I would like to see a websocket sequence diagram for all of these, but maybe not for now. It is confusiong to just see them in the code. It's hard to catch bugs.

Reply:
I'll try to make one :+1:

Reply:
Done: https://excalidraw.com/#room=49a2157c70806baa3387,r-Mkp4AvdX_K__tHj4_4Hw

Reply:
As far as I can see, 'updateGame' is sent to gameId, not to the separate userIds. What do you think about the server withholding sending the 'gameJoined' response until you have both clients (the first one joins, and waits to see a 'gameJoined', just like it does now), and when the second player sends 'joinGame' and you can have its client join the room, then send 'gameJoined' to the gameroom, at the same time?

Reply:
And what happens with the disconnects and with our game map if a user disconnects after its opponent has sent 'gameJoined' and before the disconnecting player sends 'gameJoined'?

Reply:
Otherwise you have a total of three layers of cascading wait, the third one silent and disconcerting.

Reply:
But otherwise, the wait will be with the button, and will make him/her think that the button didn't work.... mmmh

Reply:
It's okay the way it is, I guess.

Reply:
I agree that as it is now, it's a bit confusing since the users don't have a way to know the game state.

I think this will be better when we implement the pause notification.